modules/apps/commerce/commerce-order-content-web/src/main/resources/META-INF/resources/js/multi_shipping/DeliveryGroupModal.tsx

```diff
@@ -0,0 +1,246 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayModal from '@clayui/modal';
+import {Observer} from '@clayui/modal/lib/types';
+import classnames from 'classnames';
+import {sub} from 'frontend-js-web';
+import React, {useCallback, useRef, useState} from 'react';
+
+import AddressSelector from './AddressSelector';
+import ErrorMessage from './ErrorMessage';
+import {IDeliveryGroup, IFieldError, IPostalAddress} from './Types';
+
+const MANDATORY_FIELDS = ['name'];
+
+interface IDeliveryGroupModalProps {
+	onOpenModal(value: boolean): void;
+	accountId: number;
+	deliveryGroup?: IDeliveryGroup;
+	handleSubmit: any;
+	hasManageAddressesPermission?: boolean;
+	namespace?: string;
+	observerModal: Observer;
+	spritemap?: string;
+}
+
+const DeliveryGroupModal = ({
+	accountId,
+	deliveryGroup,
+	handleSubmit,
+	hasManageAddressesPermission = true,
+	namespace = 'DeliveryGroupModal',
+	observerModal,
+	onOpenModal,
+	spritemap = '',
+}: IDeliveryGroupModalProps) => {
+	const formRef = useRef(null);
+	const [deliveryGroupState, setDeliveryGroupState] =
+		useState<IDeliveryGroup>(
+			deliveryGroup || {
```

Comment:
Default in component signature: the defensive `||` has little value, as `{}` will make the left side true nonetheless.